packages/design-system/src/components/table/components/tableBody/bodyCell.tsx

```diff
@@ -17,28 +17,78 @@
 /**
  * External dependencies.
  */
-import React from 'react';
+import React, { useState, useCallback } from 'react';
+import { createPortal } from 'react-dom';
 
 /**
  * Internal dependencies.
  */
-import type { InfoType } from '../../useTable';
+import type { InfoType, TableRow } from '../../useTable';
+import { CookieTableData } from '@ps-analysis-tool/common';
 interface BodyCellProps {
   cell: React.JSX.Element | InfoType;
   width: number;
   isHighlighted?: boolean;
   isRowFocused: boolean;
+  isExtension?: boolean;
+  row: TableRow;
+  onRowClick: () => void;
 }
 
 const BodyCell = ({
+  onRowClick,
```

Comment:
Is there any reason the right-click handling is not done at the row level?

Reply:
This is because the row value displayer component is different for some columns and adding a context menu to that duplicates the efforts.